starter-code/pizza.js

```diff
@@ -17,69 +17,187 @@ var state = {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function($mushroom) {
+    if (state.mushrooms) {
+      $mushroom.style.visibility = "visible";
+    } else {
+      $mushroom.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function($greenPeppers) {
+    if (state.greenPeppers) {
+      $greenPeppers.style.visibility = "visible";
+    } else {
+      $greenPeppers.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce").forEach(function($sauce) {
+    if (state.whiteSauce) {
+      $sauce.classList.add("sauce-white");
+    } else {
+      $sauce.classList.remove("sauce-white");
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  document.querySelectorAll(".crust").forEach(function($gfc) {
+    if (state.glutenFreeCrust) {
+      $gfc.classList.add("crust-gluten-free");
+      // $gfc.style.visibility = "visible";
+    } else {
+      $gfc.classList.remove("crust-gluten-free");
+      // $gfc.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  document.querySelectorAll(".btn-pepperonni").forEach(function($btn) {
+    if (state.pepperonni) {
+      $btn.classList.add("active");
+    } else {
+      $btn.classList.remove("active");
+    }
+  });
+  document.querySelectorAll(".btn-mushrooms").forEach(function($btn) {
+    if (state.mushrooms === true) {
+      $btn.classList.add("active");
+    } else {
+      $btn.classList.remove("active");
+    }
+  });
+  document.querySelectorAll(".btn-green-peppers").forEach(function($btn) {
+    if (state.greenPeppers) {
+      $btn.classList.add("active");
+    } else {
+      $btn.classList.remove("active");
+    }
+  });
+  document.querySelectorAll(".btn-sauce").forEach(function($btn) {
+    if (state.whiteSauce) {
+      $btn.classList.add("active");
+    } else {
+      $btn.classList.remove("active");
+    }
+  });
+  document.querySelectorAll(".btn-crust").forEach(function($btn) {
+    if (state.glutenFreeCrust) {
+      $btn.classList.add("active");
+    } else {
+      $btn.classList.remove("active");
+    }
+  });
 }
 
 function renderPrice() {
+  let sum = 10;
+  document.querySelectorAll(".btn-pepperonni").forEach(function(price) {
+    if (state.pepperonni) {
+      sum += 1;
```

Comment:
If you look at the top at your ingredients object; they already have given a price for the items so there is no need to hardcode your price.